core/server/src/main/java/alluxio/worker/netty/DataServerBlockWriteHandler.java

```diff
@@ -50,9 +50,8 @@
     public BlockWriter mBlockWriter;
 
     public BlockWriteRequestInternal(Protocol.WriteRequest request) throws Exception {
-      mWorker
-          .createBlockRemote(request.getSessionId(), request.getId(), mStorageTierAssoc.getAlias(0),
-              FILE_BUFFER_SIZE);
+      mWorker.createBlockRemote(request.getSessionId(), request.getId(),
+          mStorageTierAssoc.getAlias(request.getTier()), FILE_BUFFER_SIZE);
```

Comment:
Do you need to check if request.getTier() is not null to support backward compatibility?